### ISSUE TITLE
fastlane sigh resign asks for the pp path even when it's provided in the -p option

### ISSUE THREAD

**Issue by dulgan** [fastlane/fastlane#7935]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

“fastlane sigh resign" keeps asking me the path to the provisionning profile even with the correct path passed as the -p option

I anonymized the informations below for client privacy purpose.

P
##### Complete output when running fastlane, including the stack trace and command used
fastlane sigh resign ../myipa.ipa --signing_identity "iPhone Distribution: MyOrganization (MYIDISHERE)" -p certs/AdHoc_fr.myorganization.myapp.mobileprovision --verbose
Path to provisioning file: certs/AdHoc_fr.myorganization.myapp.mobileprovision
[...]
Successfully signed ../myipa.ipa

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                 | Value                                       |
| ------------------- | ------------------------------------------- |
| OS                  | 10.11.6                                     |
| Ruby                | 2.2.3                                       |
| Bundler?            | false                                       |
| Git                 | git version 2.10.1 (Apple Git-78)           |
| Installation Source | /usr/local/bin/fastlane                     |
| Host                | Mac OS X 10.11.6 (15G1108)                  |
| Ruby Lib Dir        | /usr/local/Cellar/ruby/2.2.3/lib            |
| OpenSSL Version     | OpenSSL 1.0.2d 9 Jul 2015                   |
| Is contained        | false                                       |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version       | 8.2.1                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | fr_FR.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://docs.fastlane.tools/actions
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.9.0"

default_platform :ios

platform :ios do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
    # cocoapods
    
  end

  desc "Runs all the tests"
  lane :test do
    scan
  end

  desc "Submit a new dev build to ota"
  lane :beta do
    sigh(adhoc:true)
    gym(scheme: "myscheme", export_method: 'ad-hoc', silent: 'true', output_name: 'myschemedev', use_legacy_build_api: true)
    sh "./scripts/upload_ipa.sh"
  end

  desc "Submit a new release build to ota"
  lane :release do
    sigh(adhoc:true)
    gym(scheme: "myschemeprod", export_method: 'ad-hoc', silent: 'true', output_name: 'myschemeprod', use_legacy_build_api: true)
    sh "./scripts/upload_ipa.sh prod"
  end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://docs.fastlane.tools/actions

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer
```
</details>

<details><summary>`./Appfile`</summary>

```ruby
app_identifier "fr.myorganization.myapp" # The bundle identifier of your app
apple_id "my@apple.id" # Your Apple email address

team_id "N4PEHJ3Y7G"  # Developer Portal Team ID

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.9.0   | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| terminal-table            | 1.4.5        |
| plist                     | 3.1.0        |
| addressable               | 2.4.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander                 | 4.4.0        |
| excon                     | 0.54.0       |
| faraday                   | 0.9.2        |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20160826 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 1.6.8        |
| gh_inspector              | 1.0.2        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.1.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.2.4      |
| google-api-client         | 0.9.18       |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.5.5        |
| rubyzip                   | 1.1.7        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.1.1        |
| bundler                   | 1.13.5       |
| faraday_middleware        | 0.10.0       |
| i18n                      | 0.7.0        |
| thread_safe               | 0.3.5        |
| tzinfo                    | 1.2.2        |
| minitest                  | 5.4.3        |
| activesupport             | 4.2.7.1      |
| CFPropertyList            | 2.3.4        |
| claide                    | 1.0.1        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.1        |
</details>

*generated on:* **2017-01-20**
</details>

**Comment by mgrebenets** [comment]
@dulgan I don't know why yet, but at the moment short options don't work for some reason.
I've noticed that `-i` doesn't work too, so I suggest something's broken in option parsing logic.

Could you temporary switch to using `--provisioning_profile` instead until the issue is fixed?

**Comment by dulgan** [comment]
Thanks for looking into this, I chose to pipe the option directly for the
moment.
Thanks for the --provisionning_profile advice

**Comment by louiscremen** [comment]
Using `--provisioning_profile` worked for me.

**Comment by mfurtak** [comment]
I've put together a proposal for a fix for _sigh_'s option conflicts in #8129. If you have the time to pull that branch and let us know how it looks for you, that would be really helpful 👍
